src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -392,132 +443,137 @@ void func_80B14818(EnTa* this, PlayState* play) {
-        this->unk_2CC = 5;
-        SET_EVENTCHKINF(EVENTCHKINF_14);
+        EnTa_SetupAction(this, EnTa_RunAwayStart, EnTa_AnimRepeatCurrent);
+        this->timer = 5;
+        SET_EVENTCHKINF(EVENTCHKINF_TALON_RETURNED_FROM_CASTLE);
         Animation_PlayOnce(&this->skelAnime, &gTalonRunTransitionAnim);
         this->currentAnimation = &gTalonRunAnim;
     }
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B14C18(EnTa* this, PlayState* play) {
-    if (func_80B142F4(this, play, 0x702C)) {
-        func_80B13AA0(this, func_80B14B6C, func_80B167C0);
+void EnTa_IdleWokenInCastle(EnTa* this, PlayState* play) {
+    if (EnTa_TalkToPlayer(this, play, 0x702C)) {
+        EnTa_SetupAction(this, EnTa_TalkWokenInCastle, EnTa_AnimRepeatCurrent);
     }
-    func_80B14248(this);
+    EnTa_DecreaseShadowSize(this);
 }
 
-void func_80B14C60(EnTa* this, PlayState* play) {
+void EnTa_TalkWokenInKakariko(EnTa* this, PlayState* play) {
     if (Actor_TextboxIsClosing(&this->actor, play)) {
-        func_80B13AA0(this, func_80B14CAC, func_80B167C0);
+        EnTa_SetupAction(this, EnTa_IdleWokenInKakariko, EnTa_AnimRepeatCurrent);
     }
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B14CAC(EnTa* this, PlayState* play) {
-    if (GET_EVENTCHKINF(EVENTCHKINF_18)) {
-        if (func_80B142F4(this, play, 0x5017)) {
-            func_80B13AA0(this, func_80B14C60, func_80B167C0);
-            SET_EVENTCHKINF(EVENTCHKINF_6B);
+void EnTa_IdleWokenInKakariko(EnTa* this, PlayState* play) {
+    if (GET_EVENTCHKINF(EVENTCHKINF_EPONA_OBTAINED)) {
+        if (EnTa_TalkToPlayer(this, play, 0x5017)) {
+            EnTa_SetupAction(this, EnTa_TalkWokenInKakariko, EnTa_AnimRepeatCurrent);
+            SET_EVENTCHKINF(EVENTCHKINF_TALON_RETURNED_FROM_KAKARIKO);
         }
-    } else if (func_80B142F4(this, play, 0x5016)) {
-        func_80B13AA0(this, func_80B14C60, func_80B167C0);
+    } else if (EnTa_TalkToPlayer(this, play, 0x5016)) {
+        EnTa_SetupAction(this, EnTa_TalkWokenInKakariko, EnTa_AnimRepeatCurrent);
     }
-    func_80B14248(this);
+    EnTa_DecreaseShadowSize(this);
 }
 
-void func_80B14D4C(EnTa* this, PlayState* play) {
+void EnTa_TalkBackAtRanch(EnTa* this, PlayState* play) {
     if (Actor_TextboxIsClosing(&this->actor, play)) {
-        func_80B13AA0(this, func_80B14D98, func_80B167C0);
+        EnTa_SetupAction(this, EnTa_IdleBackAtRanch, EnTa_AnimRepeatCurrent);
     }
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B14D98(EnTa* this, PlayState* play) {
-    if (func_80B142F4(this, play, 0x2055)) {
-        func_80B13AA0(this, func_80B14D4C, func_80B167C0);
+void EnTa_IdleBackAtRanch(EnTa* this, PlayState* play) {
+    if (EnTa_TalkToPlayer(this, play, 0x2055)) {
+        EnTa_SetupAction(this, EnTa_TalkBackAtRanch, EnTa_AnimRepeatCurrent);
     }
 }
```

Comment:
do these need to say "back" I think just at ranch is probably fine no?
its more of a lore detail that he was sent back to the ranch after being somewhere else, but yea idk